src/main/java/org/breedinginsight/services/TraitUploadService.java

```diff
@@ -86,20 +90,24 @@ public ProgramUpload updateTraitUpload(UUID programId, CompletedFileUpload file,
             throw new AuthorizationException("User not in program");
         }
 
-        Optional<io.micronaut.http.MediaType> type = file.getContentType();
-        io.micronaut.http.MediaType mediaType = type.orElseThrow(() -> new UnsupportedTypeException("File upload must have a mime type"));
+        MediaType mediaType;
+        try {
+            mediaType = mimeTypeParser.getMimeType(file);
+        } catch (IOException e){
+            throw new HttpBadRequestException("Could not read file");
```

Comment:
May be a little more clear to have this as "Could not determine file type" so as to not get confused with file read errors below.

Reply:
Agreed, changed and pushed